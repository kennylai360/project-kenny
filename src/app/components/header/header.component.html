<nav>
  <div class="nav-links">
    <a class="nav-links__link__margin" routerLink="/"><i class="fa fa-home"></i><span class="nav-links__text__hide-for-mobile">Home</span></a>
    <a class="nav-links__link__margin" routerLink="/profile"><i class="fa fa-user"></i><span class="nav-links__text__hide-for-mobile">Profile</span></a>
    <a class="nav-links__link__margin" routerLink="/photography"><i class="fa fa-camera"></i><span class="nav-links__text__hide-for-mobile">Photography</span></a>
  </div>
  <div class="social-media">
    <a class="social-media__link__font-size social-media__link__margin" href="https://www.twitter.com/KENIII26" target="_blank" rel="noreferrer noopener"><i class="fa fa-twitter"></i></a>
    <a class="social-media__link__font-size social-media__link__margin" href="https://www.instagram.com/kennykinkeelai" target="_blank" rel="noreferrer noopener"><i class="fa fa-instagram"></i></a>
    <a class="social-media__link__font-size social-media__link__margin" href="https://www.twitch.tv/KENIII26" target="_blank" rel="noreferrer noopener"><i class="fa fa-twitch"></i></a>
  </div>
</nav>
