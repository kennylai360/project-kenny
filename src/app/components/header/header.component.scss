:host {
  .header-color {
    background-color: rgba(255, 255, 255, 0)
  }

  .inline-nav-ul {
    display: inline-block;
    padding-left: 5px;
  }

  .inline-blockage {
    display: inline-block;
  }

  .inline-nav {

    padding: 0px 15px 0px 0px;

    a {
      color: rgba(255, 255, 255, 0.5);
      -webkit-transition: all 0.5s ease;
      -moz-transition: all 0.5s ease;
      -ms-transition: all 0.5s ease;
      -o-transition: all 0.5s ease;
      transition: all 0.5s ease;
    }

    a:hover {
      color: rgba(255, 255, 255, 1);
      -webkit-transition: all 0.5s ease;
      -moz-transition: all 0.5s ease;
      -ms-transition: all 0.5s ease;
      -o-transition: all 0.5s ease;
      transition: all 0.5s ease;
    }
  }

  i {
    margin-right: 5px;
  }

  .link-hover {
    a {
      color: rgba(255, 255, 255, 0.5);
      -webkit-transition: all 0.5s ease;
      -moz-transition: all 0.5s ease;
      -ms-transition: all 0.5s ease;
      -o-transition: all 0.5s ease;
      transition: all 0.5s ease;
    }

    a:hover {
      color: rgba(255, 255, 255, 1);
      -webkit-transition: all 0.5s ease;
      -moz-transition: all 0.5s ease;
      -ms-transition: all 0.5s ease;
      -o-transition: all 0.5s ease;
      transition: all 0.5s ease;
    }
  }
}
