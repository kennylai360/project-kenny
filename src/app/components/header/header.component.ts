import { Component, OnInit, ViewEncapsulation } from '@angular/core';
import { ToastService } from '../../typescripts/pro/alerts/toast/toast.service';

@Component({
  selector: 'app-header',
  templateUrl: './header.component.html',
  styleUrls: ['./header.component.scss'],
  encapsulation: ViewEncapsulation.None
})
export class HeaderComponent implements OnInit {

  constructor(private toastService: ToastService) { }

  ngOnInit() {
    const ua = window.navigator.userAgent;
    if (ua.indexOf('Trident/') > -1 ||
        ua.indexOf('Edge/') > -1) {
      this.toastService.info(
        'For an optimised browsing experience, please use an another browser.',
        'Internet Explorer detected',
        {
          timeOut: 0,
          extendedTimeOut: 5000
        });
    }
  }

}
