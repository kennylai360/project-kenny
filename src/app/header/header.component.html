<!--Navbar-->
<mdb-navbar SideClass="navbar navbar-expand-lg navbar-dark fixed-top scrolling-navbar" [containerInside]="false">

<!-- Navbar brand -->
<logo><a class="navbar-brand" href="#">KennyKinKeeLai</a></logo>

<!-- Collapsible content -->
<links>
<!-- Links -->
<ul class="navbar-nav mr-auto">
<li class="nav-item" routerLinkActive="active">
<a class="nav-link waves-light" mdbRippleRadius routerLink="/home">Home<span class="sr-only">(current)</span></a>
</li>
<li class="nav-item" routerLinkActive="active">
<a class="nav-link waves-light" mdbRippleRadius routerLink="/galleries"><i class="fa fa-picture-o" aria-hidden="true"></i> Gallery</a>
</li>
</ul>
<!-- Links -->
</links>

</mdb-navbar>
<!--/.Navbar-->
