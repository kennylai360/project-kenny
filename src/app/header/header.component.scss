// add rounded border around the buttons
a {
  border-radius: 10px;
  color: white;
}

a:hover {
  color: white
}



//
//.view {
//  background-image: url("https://farm9.staticflickr.com/8502/28564464925_0a79e64a28_k.jpg")no-repeat center center;
//  background-size: cover;
//}
.navbar {
  background-color: transparent;
}
.top-nav-collapse {
  background-color: black
}
@media only screen and (max-width: 768px) {
  .navbar {
    background-color: black
  }
}
