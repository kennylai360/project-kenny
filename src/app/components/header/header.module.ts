import { NgModule, NO_ERRORS_SCHEMA } from '@angular/core';
import { CommonModule } from '@angular/common';
import { HeaderComponent } from './header.component';
import { RouterModule } from '@angular/router';
import { MDBBootstrapModule } from '../../typescripts/free/index';
import { MDBBootstrapModulePro } from '../../typescripts/pro/index';

@NgModule({
  declarations: [
    HeaderComponent
  ],
  imports: [
    CommonModule,
    RouterModule,
    MDBBootstrapModule,
    MDBBootstrapModulePro
  ],
  exports: [
    HeaderComponent
  ],
  providers: [],
  schemas: [NO_ERRORS_SCHEMA]
})
export class HeaderModule {
}
