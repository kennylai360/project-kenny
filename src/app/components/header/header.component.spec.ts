import { TestBed } from '@angular/core/testing';
import { RouterTestingModule } from '@angular/router/testing';
import { HeaderComponent } from './header.component';
import { By } from '@angular/platform-browser';

describe('HeaderComponent', () => {
  let fixture;

  beforeEach(() => {
    TestBed.configureTestingModule({
      imports: [HeaderComponent, RouterTestingModule],
    }).compileComponents();

    fixture = TestBed.createComponent(HeaderComponent);
  });

  describe('Header links', () => {
    it('should test the home link will have routerLink to /', () => {
      const homeRouterLink = fixture.debugElement
        .query(By.css('a[data-qa="qa-home-link"]'))
        .nativeElement.getAttribute('routerLink');
      expect(homeRouterLink).toEqual('/');
    });

    it('should test the resume link will have routerLink to /resume', () => {
      const resumeRouterLink = fixture.debugElement
        .query(By.css('a[data-qa="qa-resume-link"]'))
        .nativeElement.getAttribute('routerLink');
      expect(resumeRouterLink).toEqual('/resume');
    });

    it('should test the photography link will have routerLink to /photography', () => {
      const photographyRouterLink = fixture.debugElement
        .query(By.css('a[data-qa="qa-photography-link"]'))
        .nativeElement.getAttribute('routerLink');
      expect(photographyRouterLink).toEqual('/photography');
    });

    it('should test the twitter link will have href to twitter profile', () => {
      const twitterHref = fixture.debugElement
        .query(By.css('a[data-qa="qa-twitter-link"]'))
        .nativeElement.getAttribute('href');
      expect(twitterHref).toEqual('https://www.twitter.com/KENIII26');
    });

    it('should test the twitch link will have href to twitch profile', () => {
      const twitchHref = fixture.debugElement
        .query(By.css('a[data-qa="qa-twitch-link"]'))
        .nativeElement.getAttribute('href');
      expect(twitchHref).toEqual('https://www.twitch.tv/KENIII26');
    });

    it('should test the instagram link will have href to instagram profile', () => {
      const instagramHref = fixture.debugElement
        .query(By.css('a[data-qa="qa-instagram-link"]'))
        .nativeElement.getAttribute('href');
      expect(instagramHref).toEqual('https://www.instagram.com/kennykinkeelai');
    });
  });

  it('should check the UI should be consistent', () => {
    expect(fixture.nativeElement).toMatchSnapshot();
  });
});
