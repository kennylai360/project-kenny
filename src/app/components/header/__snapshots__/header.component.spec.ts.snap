// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderComponent should check the UI should be consistent 1`] = `
<div
  id="root6"
>
  <nav>
    <div
      class="nav-links"
    >
      <a
        class="nav-links__link__margin"
        data-qa="qa-home-link"
        routerlink="/"
      >
        <i
          class="fa fa-home"
        />
        <span
          class="nav-links__text__hide-for-mobile"
        >
          Home
        </span>
      </a>
      <a
        class="nav-links__link__margin"
        data-qa="qa-resume-link"
        routerlink="/resume"
      >
        <i
          class="fa fa-user"
        />
        <span
          class="nav-links__text__hide-for-mobile"
        >
          Resume
        </span>
      </a>
      <a
        class="nav-links__link__margin"
        data-qa="qa-photography-link"
        routerlink="/photography"
      >
        <i
          class="fa fa-camera"
        />
        <span
          class="nav-links__text__hide-for-mobile"
        >
          Photography
        </span>
      </a>
      <a
        class="nav-links__link__margin"
        data-qa="qa-converter-link"
        routerlink="/converter"
      >
        <i
          class="fa fa-calculator"
        />
        <span
          class="nav-links__text__hide-for-mobile"
        >
          Converter
        </span>
      </a>
    </div>
    <div
      class="social-media"
    >
      <a
        class="social-media__link__font-size social-media__link__margin"
        data-qa="qa-twitter-link"
        href="https://www.twitter.com/KENIII26"
        rel="noreferrer noopener"
        target="_blank"
      >
        <i
          class="fa fa-twitter"
        />
      </a>
      <a
        class="social-media__link__font-size social-media__link__margin"
        data-qa="qa-instagram-link"
        href="https://www.instagram.com/kennykinkeelai"
        rel="noreferrer noopener"
        target="_blank"
      >
        <i
          class="fa fa-instagram"
        />
      </a>
      <a
        class="social-media__link__font-size social-media__link__margin"
        data-qa="qa-twitch-link"
        href="https://www.twitch.tv/KENIII26"
        rel="noreferrer noopener"
        target="_blank"
      >
        <i
          class="fa fa-twitch"
        />
      </a>
    </div>
  </nav>
</div>
`;
